pkg/validator/validator_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validator
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/abcxyz/pkg/logging"
+	"github.com/abcxyz/pkg/testutil"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestValidation(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name          string
+		issuesHandler http.Handler
+		matchHandler  http.Handler
+		want          *MatchResult
+		wantErr       string
+	}{
+		{
+			name: "happy_path",
```

Comment:
Do you plan to add failure cases in this PR?

Reply:
There will be multiple failure cases, I think it is better to add them in followed-up PRs to keep current PR small

Reply:
Added and the new commit undid your approval....